internal/resource/config/authenticationapi/authentication_api_settings_data_source.go

```diff
@@ -0,0 +1,145 @@
+	apiClient      *client.APIClient
+}
+
+// Metadata returns the data source type name.
+func (r *authenticationApiSettingsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_authentication_api_settings"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *authenticationApiSettingsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+type authenticationApiSettingsDataSourceModel struct {
+	Id                               types.String `tfsdk:"id"`
+	ApiEnabled                       types.Bool   `tfsdk:"api_enabled"`
+	EnableApiDescriptions            types.Bool   `tfsdk:"enable_api_descriptions"`
+	RestrictAccessToRedirectlessMode types.Bool   `tfsdk:"restrict_access_to_redirectless_mode"`
+	IncludeRequestContext            types.Bool   `tfsdk:"include_request_context"`
+	DefaultApplicationRef            types.Object `tfsdk:"default_application_ref"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *authenticationApiSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schemaDef := schema.Schema{
+		Description: "Manages a AuthenticationApiSettings.",
+		Attributes: map[string]schema.Attribute{
+			"api_enabled": schema.BoolAttribute{
+				Description: "Enable Authentication API",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"enable_api_descriptions": schema.BoolAttribute{
+				Description: "Enable API descriptions",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"default_application_ref": schema.SingleNestedAttribute{
+				Description: "Enable API descriptions",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+				Attributes:  resourcelink.DataSourceSchema(),
+			},
+			"restrict_access_to_redirectless_mode": schema.BoolAttribute{
+				Description: "Enable restrict access to redirectless mode",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"include_request_context": schema.BoolAttribute{
+				Description: "Includes request context in API responses",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+		},
+	}
+	id.DataSourceSchema(&schemaDef, false, "The ID of this resource.")
+	resp.Schema = schemaDef
+}
+
+// Read a AuthenticationApiSettingsResponse object into the model struct
+func readAuthenticationApiSettingsResponseDataSource(ctx context.Context, r *client.AuthnApiSettings, state *authenticationApiSettingsDataSourceModel, expectedValues *authenticationApiSettingsDataSourceModel) diag.Diagnostics {
```

Comment:
Be sure to take out any unused parameters to these methods for all the data sources, like expectedValues